addons/mail/models/mail_channel.py

```diff
@@ -88,21 +88,26 @@ def _generate_random_token(self):
         ('groups', 'Selected group of users')], string='Privacy',
         required=True, default='groups',
         help='This group is visible by non members. Invisible groups can add members through the invite button.')
-    group_public_id = fields.Many2one('res.groups', string='Authorized Group',
-                                      default=lambda self: self.env.ref('base.group_user'))
+    group_public_id = fields.Many2one('res.groups', string='Authorized Group', compute='_compute_group_public_id', readonly=False, store=True)
 
     _sql_constraints = [
         ('uuid_unique', 'UNIQUE(uuid)', 'The channel UUID must be unique'),
     ]
 
-    # CHAT CONSTRAINT
+    # CONSTRAINTS
 
     @api.constrains('channel_last_seen_partner_ids', 'channel_partner_ids')
     def _constraint_partners_chat(self):
         for ch in self.sudo().filtered(lambda ch: ch.channel_type == 'chat'):
             if len(ch.channel_last_seen_partner_ids) > 2 or len(ch.channel_partner_ids) > 2:
                 raise ValidationError(_("A channel of type 'chat' cannot have more than two users."))
 
+    @api.constrains('group_public_id', 'group_ids')
+    def _constraint_group_id_channel(self):
+        for channel in self.sudo().filtered(lambda channel: channel.channel_type != 'channel'):
+            if channel.group_public_id or channel.group_ids:
```

Comment:
The `group_public_id` constraint could be made in SQL too for a stronger guarantee.

Not the other one unfortunately as it is in another table, but it's less important. If we really want, we can add a check in `_subscribe_users_automatically` to ensure the `channel_type` is as expected, but that's probably overkill?

Reply:
yes, we can do that but maybe it is not mandatory? cause this has covered most cases and the more we have the worse performance we will get?

for the group_public_id, I have the same doubt as well. do we really need the SQL constraint?

Reply:
For the other check indeed it is not really necessary if we assume data are not messed up (and as we make the upgrade to clean them it should be fine), but there is no performance issue (we are talking about adding a simple condition on a field already in cache, in comparison to a method that already iterates over all the users of a group as well as all members of the channel).

But for data integrity having SQL constraint is always better than python when possible, as this is the actual source of truth. Python constraint assume the ORM is used, but it's not always the case. And it's probably more performant to let SQL handle it rather than having to run python code (I don't really know, to be confirmed by actual benchmark if necessary, but anyway I don't think the performance issue is a concern in either case here, as creating a new channel or updating these fields is very rarely done, and also never done on a lot of channels). So the SQL one I would do definitely it now, if there was only one thing to do in this PR, it would be that one.